packages/devtools_app/lib/src/code_size/code_size_processor.dart

```diff
@@ -0,0 +1,95 @@
+// Copyright 2020 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+
+import '../charts/treemap.dart';
+import 'canned_data.dart';
+import 'code_size_controller.dart';
+
+class CodeSizeProcessor {
+  CodeSizeProcessor(this.codeSizeController);
+
+  final CodeSizeController codeSizeController;
+
+  void loadJson() {
+    final TreemapNode root = buildTreeFromJson();
+    codeSizeController.changeRoot(root);
+  }
+
+  TreemapNode addChildNode(TreemapNode parent, String name, int byteSize) {
+    TreemapNode child = parent.childrenMap[name];
+    if (child == null) {
+      child = TreemapNode(
+        name: name,
+        byteSize: byteSize,
+        childrenMap: <String, TreemapNode>{},
+      );
+      parent.childrenMap[child.name] = child;
+      parent.addChild(child);
+    } else {
+      child.byteSize += byteSize;
+    }
+    return child;
+  }
+
+  TreemapNode buildTreeFromJson() {
+    final List data = jsonDecode(sizesJson);
+
+    final rootChildren = <String, TreemapNode>{};
+
+    final TreemapNode root = TreemapNode(
+      name: 'Root',
+      childrenMap: rootChildren,
+    );
+
+    // Can optimize look up / retrieve time with a hashmap
+    for (dynamic memoryUsage in data) {
+      String libraryName = memoryUsage['l'];
```

Comment:
are these actually the empty string?

I would write a helper extension method on the Map<String, T> class that reduces the boilderplate a little bit.
```dart
extension DefaultMapGetter<String, V> on Map<String,V> {
  K lookupWithDefault(String key, V default) {
    final value = this[key];
    if (value?.isNotEmpty ?? false) return value;
    return default;
  }
}
```

Reply:
Done. I added a private helper function as well to check if `value` is a String before calling `value.isEmpty`.